### PR TITLE
Import data models from old importer-script repository

### PR DESCRIPTION
**Pull request by danduk82**
This PR adds
- data models for the geoserver types (workspace, datastore, featuretype, etc)
- a dedicated http client service with dedicated url endpoints for each resource type

**Breaking changes**
several input parameters of the GeoServerCloud class methods have been renamed:
- `workspace` -> `workspace_name`
- `datastore` -> `datastore_name`

The reason for the breaking change, is that now that we have the types, a `workspace` is an instance of the `geoservercloud.models.Workspace` class, and `workspace_name` is a string that represent the name of the workspace. This applies to all the types.

### CODE REVIEW

File: tests/test_style.py [hunk 1]
```diff
@@ -4,11 +4,79 @@
...
+                    ]
+                }
+            },
+        )
+        result = geoserver.get_styles()
+
+    assert result == ["style1", "style2"]
+
+
+@responses.activate
```
Comment (vuilleumierc) [inline tests/test_style.py:39]: really needed?

File: tests/test_feature_type.py [hunk 2]
```diff
@@ -5,6 +5,12 @@
 
 from geoservercloud.geoservercloud import GeoServerCloud
 
+# TODO: add tests for
```
[the remark this answers is not in the document]
Comment (danduk82) [reply tests/test_feature_type.py:8]: I have changed everywhere

File: tests/test_style.py (hunk 1 above)
Comment (danduk82) [reply tests/test_style.py:39]: nope

File: geoservercloud/models/datastore.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+log = logging.getLogger()
+
+
+class PostGisDataStore:
+
+    def __init__(
+        self,
+        workspace_name: str,
+        data_store_name: str,
+        connection_parameters: KeyDollarListDict,
```
[the remark this answers is not in the document]
Comment (danduk82) [reply geoservercloud/models/datastore.py:17]: done

**Review by vuilleumierc — APPROVED** [review]

**Pull request merged**